OrbitBase/PathTest.cpp

```diff
@@ -0,0 +1,196 @@
+#ifdef _WIN32
+  std::string expected = "dir1\\dir2\\dir3\\dir4\\dir5\\dir6";
+#else
+  std::string expected = "dir1\\dir2/dir3\\dir4\\/dir5\\dir6";
+#endif
+  EXPECT_EQ(actual, expected);
+}
+
+TEST(Path, JoinPathPartsAbsoluteBackslash) {
+  std::string actual = OrbitBase::JoinPath({"C:\\dir1\\dir2", "dir3\\dir4\\", "dir5\\dir6"});
+#ifdef _WIN32
+  std::string expected = "C:\\dir1\\dir2\\dir3\\dir4\\dir5\\dir6";
+#else
+  std::string expected = "C:\\dir1\\dir2/dir3\\dir4\\/dir5\\dir6";
+#endif
+  EXPECT_EQ(actual, expected);
+}
+
+TEST(Path, JoinPathPartsAbsoluteRootBackslash) {
+  std::string actual = OrbitBase::JoinPath({"C:", "dir1\\dir2", "dir3\\dir4"});
+#ifdef _WIN32
+  std::string expected = "C:dir1\\dir2\\dir3\\dir4";
+#else
+  std::string expected = "C:/dir1\\dir2/dir3\\dir4";
+#endif
+  EXPECT_EQ(actual, expected);
+}
+
+TEST(Path, FileNamesAndExtensions) {
+  std::string input = "C:\\win_abs_path\\file.txt";
+  EXPECT_EQ(OrbitBase::GetExtension(input), ".txt");
+  EXPECT_EQ(OrbitBase::StripExtension(input), "C:\\win_abs_path\\file");
+  EXPECT_EQ(OrbitBase::GetFileName(input), "file.txt");
+
+  input = "/unix_abs_path/file.txt";
+  EXPECT_EQ(OrbitBase::GetExtension(input), ".txt");
+  EXPECT_EQ(OrbitBase::StripExtension(input), "/unix_abs_path/file");
+  EXPECT_EQ(OrbitBase::GetFileName(input), "file.txt");
+
+  input = "C:\\windir.with.dots\\file_without_ext";
+  EXPECT_EQ(OrbitBase::GetExtension(input), "");
+  EXPECT_EQ(OrbitBase::StripExtension(input), "C:\\windir.with.dots\\file_without_ext");
+  EXPECT_EQ(OrbitBase::GetFileName(input), "file_without_ext");
+
+  input = "/unixdir.with.dots/file_without_ext";
+  EXPECT_EQ(OrbitBase::GetExtension(input), "");
+  EXPECT_EQ(OrbitBase::StripExtension(input), "/unixdir.with.dots/file_without_ext");
+  EXPECT_EQ(OrbitBase::GetFileName(input), "file_without_ext");
+
+  input = "relative\\mixed/path/file.txt";
+  EXPECT_EQ(OrbitBase::GetExtension(input), ".txt");
+  EXPECT_EQ(OrbitBase::StripExtension(input), "relative\\mixed/path/file");
+  EXPECT_EQ(OrbitBase::GetFileName(input), "file.txt");
+
+  input = "simple.file";
+  EXPECT_EQ(OrbitBase::GetExtension(input), ".file");
+  EXPECT_EQ(OrbitBase::StripExtension(input), "simple");
+  EXPECT_EQ(OrbitBase::GetFileName(input), "simple.file");
+
+  input = "no/ext";
+  EXPECT_EQ(OrbitBase::GetExtension(input), "");
+  EXPECT_EQ(OrbitBase::StripExtension(input), "no/ext");
+  EXPECT_EQ(OrbitBase::GetFileName(input), "ext");
+
+  input = "double.ext.txt";
+  EXPECT_EQ(OrbitBase::GetExtension(input), ".txt");
+  EXPECT_EQ(OrbitBase::StripExtension(input), "double.ext");
+  EXPECT_EQ(OrbitBase::StripExtension(OrbitBase::StripExtension(input)), "double");
+  EXPECT_EQ(OrbitBase::GetFileName(input), "double.ext.txt");
+}
+
+// TODO: A test for GetExecutableDir is missing - using this here makes this test
+// kind of invalid, since GetExecutableDir is using GetExecutablePath internally...
```

Comment:
This is a TODO but what is there to actually do? Can this be either done or the TODO removed?